serde_codegen/src/ser.rs

```diff
@@ -478,7 +516,7 @@ fn serialize_tuple_variant(
     structure_ty: P<ast::Ty>,
     fields: &[ast::StructField],
     field_names: Vec<Ident>,
```

Comment:
The field names are `__field0` `__field1` etc and they come from the captures in the switch statement that matched the current enum variant. See [ser.rs#L401](https://github.com/serde-rs/serde/blob/16d3e96b77fe8d50a2f9d9716ec231fc04251bce/serde_codegen/src/ser.rs#L401)